# HG changeset patch
# User bgrozev
# Date 1676485553 21600
#      Wed Feb 15 12:25:53 2023 -0600
# Branch 0.12
# Node ID 33da091ace9610805c5edadc3bc62c6a93d2c1c1
# Parent  9ed628635dc66da3fdc28af836efcb1a6428fc95
Set the session.ip property before the session filters are initialized when using mod_websocket

--- /usr/share/lua/5.1/prosody/core/sessionmanager.lua Thu Feb 09 22:57:32 2023 +0000
+++ /usr/share/lua/5.1/prosody/core/sessionmanager.lua Wed Feb 15 12:25:53 2023 -0600
@@ -61,7 +61,6 @@
 		end
 		return true;
 	end
-	session.ip = conn:ip();
 	local conn_name = "c2s"..tostring(session):match("[a-f0-9]+$");
 	session.log = logger.init(conn_name);
 
--- /usr/lib/prosody/modules/mod_websocket.lua Thu Feb 09 22:57:32 2023 +0000
+++ /usr/lib/prosody/modules/mod_websocket.lua Wed Feb 15 12:25:53 2023 -0600
@@ -265,15 +265,14 @@
 		return frame.data;
 	end
 
+    -- Use upstream IP if a HTTP proxy was used.
+    -- See mod_http and #540
+    conn.http_remote_ip = request.ip;
 	conn:setlistener(c2s_listener);
 	c2s_listener.onconnect(conn);
 
 	local session = sessions[conn];
 
-	-- Use upstream IP if a HTTP proxy was used
-	-- See mod_http and #540
-	session.ip = request.ip;
-
 	session.secure = consider_websocket_secure or request.secure or session.secure;
 	session.websocket_request = request;
 
--- /usr/share/lua/5.1/prosody/util/session.lua Thu Feb 09 22:57:32 2023 +0000
+++ /usr/share/lua/5.1/prosody/util/session.lua Wed Feb 15 12:25:53 2023 -0600
@@ -23,7 +23,7 @@
 
 local function set_conn(session, conn)
 	session.conn = conn;
-	session.ip = conn:ip();
+	session.ip = conn.http_remote_ip or conn:ip();
 	return session;
 end
 
